src/instructlab/utils.py

```diff
@@ -564,3 +596,53 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     if not hostname or not port:
         raise ValueError(f"Invalid host-port string: '{hostport}'")
     return hostname, port
+
+
+def convert_messages_to_legacy_dataset(
+    dataset: List[MessageSample],
+) -> List[LegacyMessageSample]:
+    """
+    Converts the new HuggingFace messages dataset format to the legacy format.
+
+    **Note**: The legacy dataset format assumes only a turn of 1. All extra turns will be dropped.
+    This means that we only look at the first 3 messages, and everything afterwards will be ignored.
+    """
+    converted_dataset: List[LegacyMessageSample] = []
+    for dp in dataset:
+        # in new dataset, the roles of a message will be determined.
+        if len(dp["messages"]) < 3:
+            raise ValueError(
+                "Dataset cannot be missing messages from roles. Please make sure that a message from each role is present."
```

Comment:
I am not a native speaker, but I think the grammar here is not ideal. "Dataset cannot be missing messages from roles" -> "Not enough messages in the dataset"?

Reply:
Thank you for the clarification, I will update this.